crates/api_structs/src/lib.rs

```diff
@@ -116,25 +116,36 @@ fn do_send_local_notifs(
   match comment.parent_id {
     Some(parent_id) => {
       if let Ok(parent_comment) = Comment::read(&conn, parent_id) {
-        if parent_comment.creator_id != user.id {
-          if let Ok(parent_user) = User_::read(&conn, parent_comment.creator_id) {
-            recipient_ids.push(parent_user.id);
-
-            if do_send_email && parent_user.send_notifications_to_email {
-              send_email_to_user(parent_user, "Reply from", "Comment Reply", &comment.content)
+        if parent_comment.creator_id != person.id {
```

Comment:
This is so users dont get notifications for their own comments, in case they reply to themselves?

Reply:
In general there are a lot of nested if here, would be clearer if you put them one after another instead, or used a helper method. Plus some comments for explanation would be nice.

Reply:
Must be to fix this issue, best leave a comment.

https://github.com/LemmyNet/lemmy-ui/issues/222#issue-832221877